test_collections/sdk_tests/support/python_testing/models/test_case.py

```diff
@@ -0,0 +1,197 @@
+                "metadata": {
+                    "public_id": identifier,
+                    "version": "0.0.1",
+                    "title": title,
+                    "description": test.description,
+                },
+            },
+        )
+
+    @staticmethod
+    def __test_identifier(name: str) -> str:
+        """Find TC-XX-1.1 in YAML title.
+        Note some have [TC-XX-1.1] and others TC-XX-1.1
+        """
+        title_pattern = re.compile(r"(?P<title>TC-[^\s\]]*)")
+        if match := re.search(title_pattern, name):
+            return match["title"]
+        else:
+            return name
+
+    @staticmethod
+    def __class_name(identifier: str) -> str:
+        """Replace all non-alphanumeric characters with _ to make valid class name."""
+        return re.sub("[^0-9a-zA-Z]+", "_", identifier)
+
+    async def setup(self) -> None:
+        logger.info("Test Setup")
+
+    async def cleanup(self) -> None:
+        logger.info("Test Cleanup")
+
+    async def execute(self) -> None:
+        try:
+            logger.info("Running Python Test: " + self.metadata["title"])
+
+            BaseManager.register("TestRunnerHooks", SDKPythonTestRunnerHooks)
+            manager = BaseManager(address=("0.0.0.0", 50000), authkey=b"abc")
+            manager.start()
+            test_runner_hooks = manager.TestRunnerHooks()  # type: ignore
+
+            runner_class = RUNNER_CLASS_PATH + RUNNER_CLASS
+            command = (
+                f"{runner_class} {self.metadata['title']}"
+                " --commissioning-method on-network --discriminator 3840 --passcode"
+                " 20202021 --storage-path /root/admin_storage.json"
+                " --paa-trust-store-path /paa-root-certs"
+            )
+
+            if self.chip_tool.pics_file_created:
+                command += f" --PICS {PICS_FILE_PATH}"
+
+            # TODO Ignoring stream from docker execution
+            self.chip_tool.send_command(
+                command,
+                prefix=EXECUTABLE,
+                is_stream=True,
+                is_socket=False,
+            ).output
+
+            while ((update := test_runner_hooks.update_test()) is not None) or (
+                not test_runner_hooks.is_finished()
+            ):
+                if not update:
+                    continue
+
+                self.__handle_update(update)
+                await sleep(0.001)
+        finally:
+            pass
+
+    def __handle_update(self, update: dict) -> None:
+        for func_name, kwargs in update.items():
+            self.__call_function_from_name(self, func_name, kwargs)
+
+    def __call_function_from_name(self, obj, func_name, kwargs) -> None:  # type: ignore
+        func = getattr(obj, func_name, None)
+        if not func:
+            raise AttributeError(f"{func_name} is not a method of {obj}")
+        if not callable(func):
+            raise TypeError(f"{func_name} is not callable")
+        func(**kwargs)
```

Comment:
I think all of this stuff is drowning a bit in here, and I would like to see if we could refactor this and move it to a separate class that takes the TestCase instance and modifies it directly instead of the `__call_function_from_name`

I think @gladystonfranca mentioned offline there was some challenges around threads and asyncio execution loops, but I would like to see if we can clean this up, so it's simpler.